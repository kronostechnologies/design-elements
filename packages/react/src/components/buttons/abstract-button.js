"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var styled_components_1 = __importDefault(require("styled-components"));
var abstract_1 = require("./styles/abstract");
var AbstractButton = styled_components_1.default.button(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  ", "\n"], ["\n  ", "\n"])), abstract_1.styles);
exports.AbstractButton = AbstractButton;
var templateObject_1;
