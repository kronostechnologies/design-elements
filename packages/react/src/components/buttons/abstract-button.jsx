import styled from 'styled-components';

import abstractStyle from './styles/abstract';

const AbstractButton = styled.button`
  ${abstractStyle}
`;

export default AbstractButton;
