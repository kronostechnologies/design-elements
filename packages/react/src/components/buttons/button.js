"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var styled_components_1 = __importDefault(require("styled-components"));
var primary_1 = require("./styles/primary");
var secondary_1 = require("./styles/secondary");
var tertiary_1 = require("./styles/tertiary");
var abstract_button_1 = require("./abstract-button");
var StyledButton = styled_components_1.default(abstract_button_1.AbstractButton)(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n    ", "\n"], ["\n    ",
    "\n"])), function (props) {
    if (props.buttonType === 'secondary') {
        return secondary_1.secondaryStyles;
    }
    else if (props.buttonType === 'tertiary') {
        return tertiary_1.tertiaryStyles;
    }
    return primary_1.primaryStyles;
});
exports.Button = function (_a) {
    var children = _a.children, disabled = _a.disabled, onClick = _a.onClick, buttonType = _a.buttonType;
    var handleClick = function () { onClick && onClick(); };
    return (<StyledButton disabled={disabled} onClick={handleClick} buttonType={buttonType}>
            {children}
        </StyledButton>);
};
var templateObject_1;
