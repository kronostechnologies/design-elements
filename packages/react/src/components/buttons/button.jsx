import React from 'react';
import styled from 'styled-components';

import { SECONDARY, TERTIARY } from '../../constants';

import primaryStyle from './styles/primary';
import secondaryStyle from './styles/secondary';
import tertiaryStyle from './styles/tertiary';

import AbstractButton from './abstract-button';

const StyledButton = styled(AbstractButton)`
  ${props => {
        if (props.type === SECONDARY) {
            return secondaryStyle;
        } else if (props.type === TERTIARY) {
            return tertiaryStyle;
        }
        return primaryStyle;
    }}
`;

const Button = ({ children, disabled, onClick }) => {
    const handleClick = () => {
        if (typeof onClick === 'function') {
            onClick();
        }
    };

    return (
        <StyledButton disabled={disabled} onClick={handleClick}>
            { children }
        </StyledButton>
    );
};

export default Button;
