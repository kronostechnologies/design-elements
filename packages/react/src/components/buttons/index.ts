export { Size } from './abstract/types';
export { ButtonProps, ButtonType, Type } from './types';
export { getBaseButtonStyles } from './abstract/styled';
export { getButtonTypeStyles, getButtonStyles } from './styled';
export { AbstractButton } from './abstract/abstract-button';
export { Button } from './button';
export { SearchButton } from './search-button';
export { IconButton } from './icon-button';
