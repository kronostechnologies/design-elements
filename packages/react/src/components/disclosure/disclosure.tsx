import { FunctionComponent, PropsWithChildren, useRef, useEffect } from 'react';
import styled from 'styled-components';
import { useId } from '../../hooks/use-id';
import {
    Size,
    Button,
    IconButton,
    ButtonProps,
} from '../buttons';
import { IconName } from '../icon/icon';

const useClickOutside = (ref: React.RefObject<HTMLElement>, handler: () => void): void => {
    useEffect(() => {
        const listener = (event: MouseEvent): void => {
            if (!ref.current || ref.current.contains(event.target as Node)) {
                return;
            }
            handler();
        };

        document.addEventListener('mousedown', listener);
        return () => {
            document.removeEventListener('mousedown', listener);
        };
    }, [ref, handler]);
};

export type ButtonPropsWithoutOnClick = Omit<ButtonProps, 'onClick'> & {
    iconName?: IconName;
    size?: Size;
}

export interface DisclosureWidgetProps {
    className?: string;
    idContent?: string;
    buttonProps: ButtonPropsWithoutOnClick;
    expanded: boolean;
    setExpanded: React.Dispatch<React.SetStateAction<boolean>>;
}

export const Container = styled.div<{ $expanded: boolean; }>`
    background-color: ${({ theme }) => theme.component['disclosure-background-color']};
    border:
        ${({
            $expanded,
            theme,
        }) => (
            $expanded ? `1px solid ${theme.component['disclosure-border-color']}` : 0
        )};
    border-radius: var(--border-radius);
    box-shadow: 0 10px 20px 0 ${({ theme }) => theme.component['disclosure-box-shadow-color']};
    color: ${({ theme }) => theme.component['disclosure-text-color']};
    display: ${({ $expanded }) => ($expanded ? 'block' : 'none')};
    position: absolute;
    z-index: 700;
`;

const DisclosureContainer = styled.div`
    position: relative;
`;

export const Disclosure: FunctionComponent<PropsWithChildren<DisclosureWidgetProps>> = ({
    className,
    idContent: providedIdContent,
    buttonProps,
    children,
    expanded,
    setExpanded,
}) => {
    const idContent = useId(providedIdContent);
    const containerRef = useRef<HTMLDivElement>(null);

    useClickOutside(containerRef, () => setExpanded(false));

    return (
        <DisclosureContainer className={className}>
            {!buttonProps.iconName && (
                <Button
                    {...buttonProps /* eslint-disable-line react/jsx-props-no-spreading */}
                    className={className}
                    onClick={() => setExpanded(!expanded)}
                    aria-expanded={expanded}
                    aria-controls={idContent}
                />
            )}
            {buttonProps.iconName && (
                <IconButton
                    {...buttonProps /* eslint-disable-line react/jsx-props-no-spreading */}
                    className={className}
                    iconName={buttonProps.iconName}
                    onClick={() => setExpanded(!expanded)}
                    aria-expanded={expanded}
                    aria-controls={idContent}
                />
            )}
            <Container $expanded={expanded} id={idContent}>
                {children}
            </Container>
        </DisclosureContainer>
    );
};
