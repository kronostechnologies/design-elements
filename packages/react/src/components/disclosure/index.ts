export { Disclosure } from './disclosure';
export type { DisclosureButtonProps, DisclosureProps } from './disclosure';
