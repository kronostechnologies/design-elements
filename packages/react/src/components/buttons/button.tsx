import React, { ReactElement } from 'react';

import styled from 'styled-components';
import { Theme } from '../theme-wrapper/theme-wrapper';
import { AbstractButton, AbstractButtonProps } from './abstract-button';

type ButtonType = 'primary' | 'secondary' | 'tertiary';

type Type = 'submit' | 'button' | 'reset';

interface ButtonProps extends AbstractButtonProps {
    /**
     * Visual style
     * @default primary
     **/
    buttonType: ButtonType;
    /**
     * Sets button type
     * @default submit
     */
    type?: Type;
}

const StyledButton = styled(AbstractButton)<ButtonProps>`
    ${(props: { theme: Theme, buttonType: ButtonType }) => {
        switch (props.buttonType) {
            case 'primary':
                return `
                    background-color: ${props.theme.main['primary-1.1']};
                    border-color: ${props.theme.main['primary-1.1']};
                    color: ${props.theme.greys.white};

                    &:hover {
                        background-color: ${props.theme.main['primary-1.3']};
                        border-color: ${props.theme.main['primary-1.3']};
                    }
                    &:focus {
                        box-shadow: 0 0 0 2px rgba(0, 128, 165, 0.4);
                    }

                    &:disabled {
                        &,
                        &:focus,
                        &:hover {
                            background-color: ${props.theme.main['primary-1.2']};
                            border-color: ${props.theme.main['primary-1.2']};
                        }
                    }
                  `;
            case 'secondary':
                return `
                    background-color: transparent;
                    border-color: ${props.theme.main['primary-1.1']};
                    color: ${props.theme.main['primary-1.1']};

                    &:hover {
                        border-color: ${props.theme.main['primary-1.3']};
                        color: ${props.theme.main['primary-1.3']};
                    }
                    &:focus {
                        box-shadow: 0 0 0 2px rgba(0, 128, 165, 0.4);
                    }

                    &:disabled {
                        &,
                        &:focus,
                        &:hover {
                            border-color: ${props.theme.main['primary-1.2']};
                            color: ${props.theme.main['primary-1.2']};
                        }
                    }
                `;
            case 'tertiary':
                return `
                    background-color: transparent;
                    border-color: transparent;
                    color: ${props.theme.greys['dark-grey']};

                    &:hover {
                        background-color: ${props.theme.greys.grey};
                        color: ${props.theme.greys.black};
                    }
                    &:focus {
                        border-color: ${props.theme.main['primary-1.1']};
                        box-shadow: 0 0 0 2px rgba(0, 128, 165, 0.4);
                    }

                    &:disabled {
                        &,
                        &:focus,
                        &:hover {
                            background-color: transparent;
                            color: ${props.theme.greys['mid-grey']};
                        }
                    }
                `;
        }
    }
}
`;

export function Button({ children, onClick, type = 'submit', ...props }: ButtonProps): ReactElement {
    function handleClick(): void {
        onClick && onClick();
    }

    const label: string = props.label || '';
    return (
        <StyledButton onClick={handleClick} type={type} {...props}>
            {children}{label}
        </StyledButton>
    );
}
