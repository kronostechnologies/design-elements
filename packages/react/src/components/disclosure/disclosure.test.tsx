import { shallow } from 'enzyme';
import { useId } from '../../hooks/use-id';
import { mountWithProviders } from '../../test-utils/renderer';
import { Button, ButtonProps } from '../buttons/button';
import { Disclosure } from './disclosure';

jest.mock('../../hooks/use-id');

type ButtonPropsWithoutOnClick = Omit<ButtonProps, 'onClick'>;

describe('Disclosure', () => {
    const content = 'content';
    const idContent = 'idContent';
    const buttonProps: ButtonPropsWithoutOnClick = {
        buttonType: 'primary',
        label: 'disclose content',
    };

    beforeEach(() => {
        jest.mocked(useId).mockReturnValue(idContent);
    });

    it('renders a button and a div', () => {
        const wrapper = mountWithProviders(<Disclosure buttonProps={buttonProps}>{content}</Disclosure>);

        expect(wrapper.find(Button).exists()).toBe(true);
        expect(wrapper.find('div').exists()).toBe(true);
    });

    it('calls useId to generated content id', () => {
        const someId = 'someId';

        shallow(<Disclosure buttonProps={buttonProps} idContent={someId}>{content}</Disclosure>);

        expect(useId).toHaveBeenCalledWith(someId);
    });

    it('sets div content id to id generated by useId', () => {
        const wrapper = mountWithProviders(
            <Disclosure buttonProps={buttonProps}>{content}</Disclosure>,
        );

        expect(wrapper.find('div').prop('id')).toBe(idContent);
    });

    it('sets button aria-controls to id generated by useId', () => {
        const wrapper = shallow(<Disclosure buttonProps={buttonProps}>{content}</Disclosure>);

        expect(wrapper.find(Button).prop('aria-controls')).toBe(idContent);
    });

    it('toggles expanded state when button is clicked', () => {
        const wrapper = mountWithProviders(
            <Disclosure buttonProps={buttonProps}>{content}</Disclosure>,
        );
        const button = wrapper.find(Button);

        expect(button.prop('aria-expanded')).toBe(false);
        button.simulate('click');

        expect(wrapper.find(Button).prop('aria-expanded')).toBe(true);
        button.simulate('click');
        expect(wrapper.find(Button).prop('aria-expanded')).toBe(false);
    });

    it('does not render children when button has not been clicked yet', () => {
        const childrenContent = <p>Test Content</p>;

        const wrapper = shallow(<Disclosure buttonProps={buttonProps}>{childrenContent}</Disclosure>);

        expect(wrapper.find('div').contains(childrenContent)).toBe(false);
    });

    it('renders children correctly after button is clicked', () => {
        const childrenContent = <p>Test Content</p>;
        const wrapper = mountWithProviders(
            <Disclosure buttonProps={buttonProps}>{childrenContent}</Disclosure>,
        );
        const button = wrapper.find(Button);

        button.simulate('click');

        expect(wrapper.find('div').exists()).toBe(true);
        expect(wrapper.find('div').contains(childrenContent)).toBe(true);
    });
});
