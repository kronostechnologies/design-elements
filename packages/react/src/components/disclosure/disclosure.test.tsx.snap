// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disclosure Button matches the snapshot when collapsed 1`] = `
.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: inherit;
  border: 1px solid;
  border-radius: 1.5rem;
  box-sizing: border-box;
  color: inherit;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 0.75rem;
  font-weight: var(--font-bold);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: 0.025rem;
  -moz-letter-spacing: 0.025rem;
  -ms-letter-spacing: 0.025rem;
  letter-spacing: 0.025rem;
  line-height: 1rem;
  min-height: var(--size-2x);
  min-width: 2rem;
  outline: none;
  padding: 0 var(--spacing-2x);
  text-transform: uppercase;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c2:focus {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c2 > svg {
  color: inherit;
  height: var(--size-1x);
  width: var(--size-1x);
}

.c1 {
  background-color: #006296;
  border-color: #006296;
  color: #FFFFFF;
}

.c1 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c1:focus {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1:hover,
.c1[aria-expanded='true'] {
  background-color: #003A5A;
  border-color: #003A5A;
  color: #FFFFFF;
}

.c1[aria-disabled='true'] {
  background-color: #84C6EA;
  border-color: #84C6EA;
  color: #FFFFFF;
  cursor: not-allowed;
}

.c3 {
  background-color: #FFFFFF;
  border: 0;
  border-radius: var(--border-radius);
  box-shadow: 0 10px 20px 0 rgb(0 0 0 / 0.2);
  color: #1B1C1E;
  display: none;
  position: absolute;
  z-index: 700;
}

.c0 {
  position: relative;
}

<Disclosure
  buttonProps={
    {
      "buttonType": "primary",
      "label": "disclose content",
    }
  }
>
  <styled.div>
    <div
      className="c0"
    >
      <Button
        aria-controls="idContent"
        aria-expanded={false}
        buttonType="primary"
        label="disclose content"
        onBlur={[Function]}
        onClick={[Function]}
      >
        <Styled(AbstractButton)
          aria-controls="idContent"
          aria-expanded={false}
          buttonType="primary"
          focusable={true}
          isMobile={false}
          onBlur={[Function]}
          onClick={[Function]}
          type="button"
        >
          <AbstractButton
            aria-controls="idContent"
            aria-expanded={false}
            buttonType="primary"
            className="c1"
            focusable={true}
            isMobile={false}
            onBlur={[Function]}
            onClick={[Function]}
            type="button"
          >
            <styled.button
              $focusable={true}
              aria-controls="idContent"
              aria-expanded={false}
              buttonType="primary"
              className="c1"
              isMobile={false}
              onBlur={[Function]}
              onClick={[Function]}
              type="button"
            >
              <button
                aria-controls="idContent"
                aria-expanded={false}
                className="c2 c1"
                onBlur={[Function]}
                onClick={[Function]}
                type="button"
              >
                disclose content
              </button>
            </styled.button>
          </AbstractButton>
        </Styled(AbstractButton)>
      </Button>
      <styled.div
        $expanded={false}
        id="idContent"
      >
        <div
          className="c3"
          id="idContent"
        >
          content
        </div>
      </styled.div>
    </div>
  </styled.div>
</Disclosure>
`;

exports[`Disclosure Button matches the snapshot when expanded 1`] = `
.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: inherit;
  border: 1px solid;
  border-radius: 1.5rem;
  box-sizing: border-box;
  color: inherit;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 0.75rem;
  font-weight: var(--font-bold);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: 0.025rem;
  -moz-letter-spacing: 0.025rem;
  -ms-letter-spacing: 0.025rem;
  letter-spacing: 0.025rem;
  line-height: 1rem;
  min-height: var(--size-2x);
  min-width: 2rem;
  outline: none;
  padding: 0 var(--spacing-2x);
  text-transform: uppercase;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c2:focus {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c2 > svg {
  color: inherit;
  height: var(--size-1x);
  width: var(--size-1x);
}

.c1 {
  background-color: #006296;
  border-color: #006296;
  color: #FFFFFF;
}

.c1 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c1:focus {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1:hover,
.c1[aria-expanded='true'] {
  background-color: #003A5A;
  border-color: #003A5A;
  color: #FFFFFF;
}

.c1[aria-disabled='true'] {
  background-color: #84C6EA;
  border-color: #84C6EA;
  color: #FFFFFF;
  cursor: not-allowed;
}

.c3 {
  background-color: #FFFFFF;
  border: 1px solid #878F9A;
  border-radius: var(--border-radius);
  box-shadow: 0 10px 20px 0 rgb(0 0 0 / 0.2);
  color: #1B1C1E;
  display: block;
  position: absolute;
  z-index: 700;
}

.c0 {
  position: relative;
}

<Disclosure
  buttonProps={
    {
      "buttonType": "primary",
      "label": "disclose content",
    }
  }
>
  <styled.div>
    <div
      className="c0"
    >
      <Button
        aria-controls="idContent"
        aria-expanded={true}
        buttonType="primary"
        label="disclose content"
        onBlur={[Function]}
        onClick={[Function]}
      >
        <Styled(AbstractButton)
          aria-controls="idContent"
          aria-expanded={true}
          buttonType="primary"
          focusable={true}
          isMobile={false}
          onBlur={[Function]}
          onClick={[Function]}
          type="button"
        >
          <AbstractButton
            aria-controls="idContent"
            aria-expanded={true}
            buttonType="primary"
            className="c1"
            focusable={true}
            isMobile={false}
            onBlur={[Function]}
            onClick={[Function]}
            type="button"
          >
            <styled.button
              $focusable={true}
              aria-controls="idContent"
              aria-expanded={true}
              buttonType="primary"
              className="c1"
              isMobile={false}
              onBlur={[Function]}
              onClick={[Function]}
              type="button"
            >
              <button
                aria-controls="idContent"
                aria-expanded={true}
                className="c2 c1"
                onBlur={[Function]}
                onClick={[Function]}
                type="button"
              >
                disclose content
              </button>
            </styled.button>
          </AbstractButton>
        </Styled(AbstractButton)>
      </Button>
      <styled.div
        $expanded={true}
        id="idContent"
      >
        <div
          className="c3"
          id="idContent"
        >
          content
        </div>
      </styled.div>
    </div>
  </styled.div>
</Disclosure>
`;

exports[`Disclosure IconButton matches the snapshot when collapsed 1`] = `
.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: inherit;
  border: 1px solid;
  border-radius: 1.5rem;
  box-sizing: border-box;
  color: inherit;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 0.75rem;
  font-weight: var(--font-bold);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: 0.025rem;
  -moz-letter-spacing: 0.025rem;
  -ms-letter-spacing: 0.025rem;
  letter-spacing: 0.025rem;
  line-height: 1rem;
  min-height: var(--size-2x);
  min-width: 2rem;
  outline: none;
  padding: 0 var(--spacing-2x);
  text-transform: uppercase;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c2:focus {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c2 > svg {
  color: inherit;
  height: var(--size-1x);
  width: var(--size-1x);
}

.c1 {
  background-color: #006296;
  border-color: #006296;
  color: #FFFFFF;
  padding: 0;
  width: var(--size-2x);
}

.c1 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c1:focus {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1:hover,
.c1[aria-expanded='true'] {
  background-color: #003A5A;
  border-color: #003A5A;
  color: #FFFFFF;
}

.c1[aria-disabled='true'] {
  background-color: #84C6EA;
  border-color: #84C6EA;
  color: #FFFFFF;
  cursor: not-allowed;
}

.c1 > svg {
  height: var(--size-1x);
  width: var(--size-1x);
}

.c3 {
  background-color: #FFFFFF;
  border: 0;
  border-radius: var(--border-radius);
  box-shadow: 0 10px 20px 0 rgb(0 0 0 / 0.2);
  color: #1B1C1E;
  display: none;
  position: absolute;
  z-index: 700;
}

.c0 {
  position: relative;
}

<Disclosure
  buttonProps={
    {
      "buttonType": "primary",
      "iconName": "home",
      "label": "disclose content",
    }
  }
>
  <styled.div>
    <div
      className="c0"
    >
      <IconButton
        aria-controls="idContent"
        aria-expanded={false}
        buttonType="primary"
        iconName="home"
        label="disclose content"
        onBlur={[Function]}
        onClick={[Function]}
      >
        <Styled(AbstractButton)
          aria-controls="idContent"
          aria-expanded={false}
          aria-label="disclose content"
          buttonType="primary"
          focusable={true}
          isMobile={false}
          onBlur={[Function]}
          onClick={[Function]}
          size="medium"
          type="submit"
        >
          <AbstractButton
            aria-controls="idContent"
            aria-expanded={false}
            aria-label="disclose content"
            buttonType="primary"
            className="c1"
            focusable={true}
            isMobile={false}
            onBlur={[Function]}
            onClick={[Function]}
            size="medium"
            type="submit"
          >
            <styled.button
              $focusable={true}
              aria-controls="idContent"
              aria-expanded={false}
              aria-label="disclose content"
              buttonType="primary"
              className="c1"
              isMobile={false}
              onBlur={[Function]}
              onClick={[Function]}
              size="medium"
              type="submit"
            >
              <button
                aria-controls="idContent"
                aria-expanded={false}
                aria-label="disclose content"
                className="c2 c1"
                onBlur={[Function]}
                onClick={[Function]}
                size="medium"
                type="submit"
              >
                <Icon
                  aria-hidden="true"
                  name="home"
                  size="16"
                >
                  <svg
                    aria-hidden="true"
                    color="currentColor"
                    focusable={false}
                    height="16"
                    width="16"
                  />
                </Icon>
              </button>
            </styled.button>
          </AbstractButton>
        </Styled(AbstractButton)>
      </IconButton>
      <styled.div
        $expanded={false}
        id="idContent"
      >
        <div
          className="c3"
          id="idContent"
        >
          content
        </div>
      </styled.div>
    </div>
  </styled.div>
</Disclosure>
`;

exports[`Disclosure IconButton matches the snapshot when expanded 1`] = `
.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: inherit;
  border: 1px solid;
  border-radius: 1.5rem;
  box-sizing: border-box;
  color: inherit;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 0.75rem;
  font-weight: var(--font-bold);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: 0.025rem;
  -moz-letter-spacing: 0.025rem;
  -ms-letter-spacing: 0.025rem;
  letter-spacing: 0.025rem;
  line-height: 1rem;
  min-height: var(--size-2x);
  min-width: 2rem;
  outline: none;
  padding: 0 var(--spacing-2x);
  text-transform: uppercase;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c2:focus {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c2 > svg {
  color: inherit;
  height: var(--size-1x);
  width: var(--size-1x);
}

.c1 {
  background-color: #006296;
  border-color: #006296;
  color: #FFFFFF;
  padding: 0;
  width: var(--size-2x);
}

.c1 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c1:focus {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1:hover,
.c1[aria-expanded='true'] {
  background-color: #003A5A;
  border-color: #003A5A;
  color: #FFFFFF;
}

.c1[aria-disabled='true'] {
  background-color: #84C6EA;
  border-color: #84C6EA;
  color: #FFFFFF;
  cursor: not-allowed;
}

.c1 > svg {
  height: var(--size-1x);
  width: var(--size-1x);
}

.c3 {
  background-color: #FFFFFF;
  border: 1px solid #878F9A;
  border-radius: var(--border-radius);
  box-shadow: 0 10px 20px 0 rgb(0 0 0 / 0.2);
  color: #1B1C1E;
  display: block;
  position: absolute;
  z-index: 700;
}

.c0 {
  position: relative;
}

<Disclosure
  buttonProps={
    {
      "buttonType": "primary",
      "iconName": "home",
      "label": "disclose content",
    }
  }
>
  <styled.div>
    <div
      className="c0"
    >
      <IconButton
        aria-controls="idContent"
        aria-expanded={true}
        buttonType="primary"
        iconName="home"
        label="disclose content"
        onBlur={[Function]}
        onClick={[Function]}
      >
        <Styled(AbstractButton)
          aria-controls="idContent"
          aria-expanded={true}
          aria-label="disclose content"
          buttonType="primary"
          focusable={true}
          isMobile={false}
          onBlur={[Function]}
          onClick={[Function]}
          size="medium"
          type="submit"
        >
          <AbstractButton
            aria-controls="idContent"
            aria-expanded={true}
            aria-label="disclose content"
            buttonType="primary"
            className="c1"
            focusable={true}
            isMobile={false}
            onBlur={[Function]}
            onClick={[Function]}
            size="medium"
            type="submit"
          >
            <styled.button
              $focusable={true}
              aria-controls="idContent"
              aria-expanded={true}
              aria-label="disclose content"
              buttonType="primary"
              className="c1"
              isMobile={false}
              onBlur={[Function]}
              onClick={[Function]}
              size="medium"
              type="submit"
            >
              <button
                aria-controls="idContent"
                aria-expanded={true}
                aria-label="disclose content"
                className="c2 c1"
                onBlur={[Function]}
                onClick={[Function]}
                size="medium"
                type="submit"
              >
                <Icon
                  aria-hidden="true"
                  name="home"
                  size="16"
                >
                  <svg
                    aria-hidden="true"
                    color="currentColor"
                    focusable={false}
                    height="16"
                    width="16"
                  />
                </Icon>
              </button>
            </styled.button>
          </AbstractButton>
        </Styled(AbstractButton)>
      </IconButton>
      <styled.div
        $expanded={true}
        id="idContent"
      >
        <div
          className="c3"
          id="idContent"
        >
          content
        </div>
      </styled.div>
    </div>
  </styled.div>
</Disclosure>
`;
